cmd/subnetcmd/helpers.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/cmd/flags"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/utils"
+	"golang.org/x/exp/slices"
+)
+
+func fillNetworkDetails(network *models.Network) error {
+	if network.Kind == models.Devnet && network.Endpoint == "" {
+		endpoint, err := app.Prompt.CaptureString("Devnet Network Endpoint")
+		if err != nil {
+			return err
+		}
+		network.Endpoint = endpoint
+	}
+	return nil
+}
+
+func GetNetworkFromCmdLineFlags(
+	useLocal bool,
+	useDevnet bool,
+	useFuji bool,
+	useMainnet bool,
+	endpoint string,
+	supportedNetworkKinds []models.NetworkKind,
+) (models.Network, error) {
+	// get network from flags
+	network := models.UndefinedNetwork
+	switch {
+	case useLocal:
+		network = models.LocalNetwork
+	case useDevnet:
+		network = models.DevnetNetwork
+	case useFuji:
+		network = models.FujiNetwork
+	case useMainnet:
+		network = models.MainnetNetwork
+	}
+
+	if endpoint != "" {
```

Comment:
no need to pass endpoint if you can implement comment above and func that can return network struct with endpoint set from IP

Reply:
the endpoint is when passed as cmdline flag, it also enables the user to pass the port in flag

Reply:
this function returns a network from cmdline flags